### ISSUE TITLE
Class-wide user-defined literals

### ISSUE THREAD

**Issue by sttaft** [Ada-Rapporteur-Group/User-Community-Input#5]
We currently don't allow the use of literals for a class-wide type even if the root type has an Integer_Literal, Real_Literal, or String_Literal aspect.  It seems like a natural generalization.

Working AI: https://docs.google.com/document/d/1Ak89RSqxTXr1vNSiLFSWZBixxDPM1e-49z9sGkeeCr4/

**Comment by ARG-Editor** [comment]
Some of us were hoping for a full proposal, especially as the rules requiring literals to be nonoverriddable seem unnecessary unless class-wide literals are supported.

    Randy.

**Comment by mosteo** [comment]
I did some experiments related to this and the aggregate initialization aspects. In case anyone is interested, there are slides and a presentation here: https://fosdem.org/2022/schedule/event/ada_2022_json_like/ and the code is here: https://github.com/mosteo/yeison

**Comment by mosteo** [comment]
Hi there, I was just reviewing the last ARG minutes and this and #61 were assigned to me. Could someone with permissions add me to the Assignees of the issues?

@sttaft what would be the first steps? draft an IA? come up with some examples that clarify the issue at hand?

**Comment by sttaft** [comment]
@mosteo Yes, you should draft an AI using Google Docs, in the New AIs folder, and include one or more examples in the !issue and the !examples sections.

New AIs folder is here:

  https://drive.google.com/drive/folders/1tNeRkP7uznWDatkdVkFs8reKddMqoTxB?usp=sharing

Start by copying the AI22-0000-template from the Active AIs folder down into the New AIs folder, and then edit the new file following the instructions.

Active AIs folder is here:

   https://drive.google.com/drive/folders/1UVfIIjatXhs5suvTxrOQzThlCGbl4rQq?usp=sharing

If you have questions that come up while drafting the AI, add a comment to this GitHub issue and hopefully someone will reply!

As far as being assigned the issue, I don't seem to have the ability to do that.  Randy and Richard Wai are the only administrators of this GitHub repository, as far as I know.

**Comment by Richard-Wai** [comment]
@mosteo I have sent you an invite

**Comment by ARG-Editor** [comment]
This topic was addressed in AI22-0109-1, which was approved by ARG meeting #63E. So this issue is completed.
